drivers: can: mcux: flexcan: support Enhanced CAN Bit Timing registers

william-tang914 opened the pull request:
Some FlexCAN devices have enhanced bit timing registers (EPRS ENCBT EDCBT) with extended timing configuration ranges and different behavior compared to standard bit timing registers.

This change adds proper support for these enhanced registers, utilize full bit timing capabilities.

Key differences for enhanced bit timing registers:
- Extended timing ranges with larger maximum values for sjw, phase_seg1, phase_seg2, and prescaler fields
- No propagation segment configuration support, so prop_seg must be 0
- prop_seg value is used directly without the typical 'minus 1' adjustment

The implementation adds conditional compilation based on the `FSL_FEATURE_FLEXCAN_HAS_ENHANCED_BIT_TIMING_REG` feature flag. For Enhanced registers, use extended timing limits and prop_seg directly.

Test this commit on mimxrt1180_evk/mimxrt1189/cm33 drivers.can.timing test case, and enable `TEST_ALL_BITRATES` Kconfig symbol.

william-tang914 committed:
drivers: can: mcux: flexcan: support Enhanced CAN Bit Timing registers

Some FlexCAN devices have enhanced bit timing registers (EPRS ENCBT EDCBT)
with extended timing configuration ranges and different behavior compared
to standard bit timing registers.

This change adds proper support for these enhanced registers, utilize full
bit timing capabilities.

Key differences for enhanced bit timing registers:
- Extended timing ranges with larger maximum values for sjw, phase_seg1,
  phase_seg2, and prescaler fields
- No propagation segment configuration support, so prop_seg must be 0
- prop_seg value is used directly without the typical 'minus 1' adjustment

The implementation adds conditional compilation based on the
`FSL_FEATURE_FLEXCAN_HAS_ENHANCED_BIT_TIMING_REG` feature flag.
For Enhanced registers, use extended timing limits and prop_seg directly.

Test this commit on mimxrt1180_evk/mimxrt1189/cm33 drivers.can.timing
test case, and enable `TEST_ALL_BITRATES` Kconfig symbol.

Signed-off-by: William Tang <william.tang@nxp.com>

henrikbrixandersen reviewed (changes requested):
> Test this commit on mimxrt1180_evk/mimxrt1189/cm33 drivers.can.timing test case, and enable TEST_ALL_BITRATES Kconfig symbol.

Looks good, but the above seems to be missing from the commits?

william-tang914 commented:
Add `CONFIG_TEST_ALL_BITRATES=y` for mimxrt1180_evk/mimxrt1189/cm33.

william-tang914 committed:
tests: drivers: can: timing: enable full timing test on mimxrt1180_evk

Enable the full range of CAN timing tests on the NXP mimxrt1180_evk
board cm33 core.

Signed-off-by: William Tang <william.tang@nxp.com>
